src/transformers/models/align/processing_align.py

```diff
@@ -41,12 +45,59 @@ class AlignProcessor(ProcessorMixin):
 
     def __init__(self, image_processor, tokenizer):
         super().__init__(image_processor, tokenizer)
-
-    def __call__(self, text=None, images=None, padding="max_length", max_length=64, return_tensors=None, **kwargs):
+        self.processing_kwargs: ProcessingKwargs = {
+            "common_kwargs": {"return_tensors": None},
+            "text_kwargs": {
+                "text_pair": None,
+                "text_target": None,
+                "text_pair_target": None,
+                "add_special_tokens": True,
+                "padding": "max_length",
+                "truncation": True,
+                "max_length": 64,
+                "stride": 0,
+                "is_split_into_words": False,
+                "pad_to_multiple_of": None,
+                "return_token_type_ids": None,
+                "return_attention_mask": None,
+                "return_overflowing_tokens": False,
+                "return_special_tokens_mask": False,
+                "return_offsets_mapping": False,
+                "return_length": False,
+                "verbose": True,
+            },
+            "images_kwargs": {
+                "do_crop_margin": None,
+                "do_resize": None,
+                "size": None,
+                "resample": None,
+                "do_thumbnail": None,
+                "do_align_long_axis": None,
+                "do_pad": None,
+                "do_rescale": None,
+                "rescale_factor": None,
+                "do_normalize": None,
+                "image_mean": None,
+                "image_std": None,
+                "data_format": "channels_first",
+                "input_data_format": None,
+            },
+            "audio_kwargs": {},
+            "videos_kwargs": {},
+        }
+
+    def __call__(
+        self,
+        text: Union[TextInput, PreTokenizedInput, List[TextInput], List[PreTokenizedInput]] = None,
+        images: ImageInput = None,
+        audio=None,
+        videos=None,
+        **kwargs,
```

Comment:
My main question is how are type hints and signature scan done for this? Is there a way to say that the kwargs include text kwargs and image kwargs + benefit from having the doc as well?